server/issue.go

```diff
@@ -513,8 +514,21 @@ func (s *Server) createPipelineValidateOnly(ctx context.Context, pc *api.Pipelin
 			PipelineID:    sc.PipelineID,
 			EnvironmentID: sc.EnvironmentID,
 		}
-		for _, tc := range sc.TaskList {
+		// We don't know IDs before inserting, so we use array index instead.
+		// indexBlockedByIndex[indexA] holds indexes of the tasks that block taskList[indexA]
```

Comment:
IIUC, this is not guaranteed the task id would be sequential. what if there is a concurrent task creation, would the task id be interleaved?

Reply:
I think the task IDs in the same stage are gauranteed to be sequential here. Note we are in function `createPipelineValidateOnly`, so there are no tasks inserted.